server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -297,6 +309,16 @@ public void addNamedQuery(String name, Query query) {
         }
     }
 
+    public void addQueryUsage(String query) {
+        usedQueries.add(query);
+    }
+
+    public void reportQueriesUsage() {
+        for (String query : usedQueries) {
+            searchUsageService.incrementQueryUsage(query);
+        }
+    }
```

Comment:
I had a deeper look also at the existing search stats: they were originally made to be returned from nodes stats, then a subset of it was also exposed by cluster stats (but the search stats are not part of it). It looks like though there is existing infra to report on shard level things as part of cluster stats.

That makes me wonder: even if we collect at the shard level, does it make sense to expose these additional counters at the node level (node feature usage)? Does it add value to be able to distinguish nodes from one another? Would it make more sense to expose these aggregated through cluster stats? Would that simplify how the data is consumed,  or would that make it harder, or aggregating on the client side is easy enough? 

It would seem as moving the reporting from one API to another should not be a huge job, provided that the collection stays unchanged.